src/pspm_struct2vec.m

```diff
@@ -0,0 +1,42 @@
+function v = pspm_struct2vec(S, field, warningtype)
+% ● Description
+%   pspm_struct2vec turns a numerical field in a multi-element structure
+%   array into a numerical vector. If in every element of the structure 
+%   array, the field has one element, this returns the same output as
+%   [S(:).field]. If fields ar empty or have more than one element, the
+%   output vector will be made to have the same number of elements as S,
+%   and a warning will be thrown. 
+% ● Format
+%   v = pspm_struct2vec(S, field, warningtype)
+% ● Arguments
+%   *           S : a structure array.
+%   *       field : name of a numerical field.
+%   * warningtype : ['marker' or 'generic'] Type of warning to the displayed
+%                   for the user.
+% ● Output
+%   *     v : numerical vector.
+% ● History
+%   Introduced in PsPM 7.0
+%   Written in 2024 Dominik R Bach (Uni Bonn)
+
+if nargin < 3 || strcmpi(warningtype, 'generic')
+    warningstr = 'Element';
+else
+    warningstr = 'Marker value';
+end
+
+% for loops are faster than arrayfun
+n = numel(S);
```

Comment:
I tested this function with two-dimensional structure array, and I can see it works well with this function. The only thing is, this function will provide a 1xN vector `v`, where `N` denotes the number of elements in the structure array. If it is a two dimensional structure array with size `p x q`, the output will be `1 x p*q` vector.

Reply:
Testing code:

``
field = 'f';
value = {'some text','a';
[10, 20, 30],'b';
magic(5),3};
S = struct(field,value);
v = pspm_struct2vec(S, 'f')